### PR TITLE
PLANES

### PR DESCRIPTION
**Pull request by ghost**
Why did I do this? Cluster said that it needed done one day by someone. I had nothing better to do today.

Adds a few new planes and put their respective stuff in them. Pretty straightforward. A move away from layers nonsense.

- PLANE_TURF
- PLANE_NOIR_BLOOD
- PLANE_OBJ
- PLANE_MOB
- PLANE_EFFECTS
- PLANE_LIGHTING 
- PLANE_STATIC
- PLANE_HUD

Is this tested? like, at all?
yes. I can't guarantee there are no bugs but after a day of testing I fixed all the bugs I found. See the comments for proof.

### CODE REVIEW

**Comment by ghost** [comment]
conflicts on arrival is always a good sign

File: code/game/dna/genes/vg_powers.dm [hunk 1]
```diff
@@ -121,6 +121,20 @@ Obviously, requires DNA2.
...
+// NOIR
+
+/obj/screen/plane_master/noir_master
+	plane = PLANE_NOIR_BLOOD
+	color = list(1,0,0,0,
+				 0,1,0,0,
+				 0,0,1,0,
+				 0,0,0,0,
+				 0,0,0,1)
+	appearance_flags = NO_CLIENT_COLOR
```
Comment (ghost) [inline code/game/dna/genes/vg_powers.dm]: this doesn't work by the way @clusterfack - not sure exactly why.

**Comment by ghost** [comment]
compile error:
vgstation13.mdme:14:error: FLOAT_PLANE: undefined var
poor little guy

**Comment by ghost** [comment]
compiles on my machine (tm) - is the compiler not at 510 or something? Pretty sure FLOAT_PLANE is a byond var.

File: code/modules/overlays/mobs.dm [hunk 2]
```diff
@@ -1,4 +1,8 @@
 // Human Overlay objects for the new Overlays system.
+
+/obj/Overlays
```
Comment (ghost) [inline code/modules/overlays/mobs.dm]: To do: undo this change it didn't work anyway

File: code/modules/projectiles/guns/hookshot.dm [hunk 3]
```diff
@@ -318,10 +318,13 @@
 
 /obj/effect/overlay/chain/update_icon()
 	overlays.len = 0
+	var/image/chain_img
 	if(extremity_A && (loc != extremity_A.loc))
-		overlays += image(icon,src,"chain",MOB_LAYER-0.1,get_dir(src,extremity_A))
+		chain_img = image(icon,src,"chain",MOB_LAYER-0.1,get_dir(src,extremity_A))
 	if(extremity_B && (loc != extremity_B.loc))
-		overlays += image(icon,src,"chain",MOB_LAYER-0.1,get_dir(src,extremity_B))
+		overlays = image(icon,src,"chain",MOB_LAYER-0.1,get_dir(src,extremity_B))
```
Comment (ghost) [inline code/modules/projectiles/guns/hookshot.dm]: Oh dear change to chain_img

**Comment by clusterfack** [comment]
We may not have set the compiler to the proper minor version. Modify travis.yml.

File: code/game/dna/genes/vg_powers.dm (hunk 1 above)
Comment (clusterfack) [inline code/game/dna/genes/vg_powers.dm]: You should read your code again and be ashamed.

*also that piss poor copypaste smh*

**Comment by Iamgoofball** [comment]
do you like gladiator planes

File: code/game/dna/genes/vg_powers.dm (hunk 1 above)
Comment (ghost) [inline code/game/dna/genes/vg_powers.dm]: I didn't even copy paste it.

Comment (DrCelt) [inline code/game/dna/genes/vg_powers.dm]: Though I was a little drunk.

Comment (ghost) [inline code/game/dna/genes/vg_powers.dm]: Still can't see it.

File: code/game/objects/structures/vehicles/adminbus.dm [hunk 4]
```diff
@@ -35,19 +36,19 @@
...
 	src.dir = EAST
 	playsound(src, 'sound/misc/adminbus.ogg', 50, 0, 0)
 	lightsource = new/obj/structure/buslight(src.loc)
 	update_lightsource()
 	warp = new/obj/structure/teleportwarp(src.loc)
 	busjuke = new/obj/machinery/media/jukebox/superjuke/adminbus(src.loc)
+	busjuke.plane = PLANE_EFFECTS
 	busjuke.dir = EAST
-	layer = FLY_LAYER+1
-	spawn(10)
```
Comment (ghost) [inline code/game/objects/structures/vehicles/adminbus.dm:49]: What the hell @deitylink

File: code/game/dna/genes/vg_powers.dm (hunk 1 above)
Comment (clusterfack) [inline code/game/dna/genes/vg_powers.dm]:
```
appearance_flags = NO_CLIENT_COLOR
```

Comment (ghost) [inline code/game/dna/genes/vg_powers.dm]: ?

Comment (PJB3005) [inline code/game/dna/genes/vg_powers.dm]: You haven't given it the plane master appearance flag.

Comment (ghost) [inline code/game/dna/genes/vg_powers.dm]: I am a buck eejit

**Comment by ghost** [comment]
with that commit I've fixed all the bugs I know of. I'll now do a tonne of testing.

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 5]
```diff
@@ -42,9 +42,11 @@
 
 /obj/structure/bed/chair/proc/handle_layer()
 	if(dir == NORTH)
-		src.layer = FLY_LAYER
+		layer = FLY_LAYER
+		plane = PLANE_EFFECTS
 	else
-		src.layer = OBJ_LAYER
```
Comment (ghost) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm:47]: take note @9600bauds

File: code/modules/mob/living/carbon/alien/larva/powers.dm [hunk 6]
```diff
@@ -9,9 +9,11 @@
 
 	if(layer != TURF_LAYER+0.2)
 		layer = TURF_LAYER+0.2
+		plane = TURF_LAYER
```
Comment (ghost) [inline code/modules/mob/living/carbon/alien/larva/powers.dm]: Bug found here. Should be PLANE_TURF - this actually caused aliens when hiding to be visible beyond the darkness.

File: code/modules/mob/living/say.dm [hunk 7]
```diff
@@ -410,6 +410,7 @@ var/list/department_radio_keys = list(
 			speech_bubble_recipients.Add(M.client)
 	spawn(0)
 		var/image/speech_bubble = image('icons/mob/talk.dmi', get_holder_at_turf_level(src), "h[bubble_type][say_test(message)]",MOB_LAYER+1)
+		speech_bubble.plane = PLANE_EFFECTS
```
Comment (ghost) [inline code/modules/mob/living/say.dm]: Should I make this above the darkness? @PJB3005 - what do you think?

File: code/modules/mob/living/silicon/ai/freelook/chunk.dm [hunk 8]
```diff
@@ -114,8 +114,9 @@
 		var/turf/t = turf
 		if(obscuredTurfs[t])
 			if(!t.obscured)
-				t.obscured = image('icons/effects/cameravis.dmi', t, "black", 15)
-
+				var/image/obscured_static = image('icons/effects/cameravis.dmi', t, "black", 15)
```
Comment (ghost) [inline code/modules/mob/living/silicon/ai/freelook/chunk.dm:117]: Whoever hardcoded the layer, I hate you.

File: code/modules/mob/living/silicon/mommi/update_icons.dm [hunk 9]
```diff
@@ -37,7 +37,9 @@
 	var/overlay_layer = LIGHTING_LAYER + 1
 	if(layer != MOB_LAYER)
 		overlay_layer=TURF_LAYER+0.2
-	overlays += image(icon,"eyes-[subtype][emagged?"-emagged":""]",overlay_layer)
+	var/image/eyes = image(icon,"eyes-[subtype][emagged?"-emagged":""]",overlay_layer)
+	eyes.plane = PLANE_LIGHTING
```
Comment (ghost) [inline code/modules/mob/living/silicon/mommi/update_icons.dm]: Bug found here: if you hide the eyes remain at plane lighting.

File: code/modules/mob/living/say.dm (hunk 7 above)
Comment (PJB3005) [inline code/modules/mob/living/say.dm]: I don't think so, personally.

Comment (ghost) [inline code/modules/mob/living/say.dm]: Alright.

Comment (PJB3005) [inline code/modules/mob/living/say.dm]: Actually when I think about it, it'd be neat.

**Comment by ghost** [comment]
2 other bugs found:
ever since rebasing, objects in hand do not up their plane correctly (Edit: This has been fixed)
objects in containers do not appear above the hud. (Edit: This has been fixed)

**Comment by ghost** [comment]
build mode hud does not appear correctly (Edit: This has been fixed)

**Comment by ghost** [comment]
mommi/borg modules do not appear correctly in the hud  (Edit: This has been fixed)

**Comment by ghost** [comment]
Objects dropped by MoMMIs do not return to the correct plane.  (Edit: This has been fixed)

File: code/modules/mob/living/say.dm (hunk 7 above)
Comment (ghost) [inline code/modules/mob/living/say.dm]: oh, okay then.

**Comment by PJB3005** [comment]
This is the kind of testing every PR needs.

**Comment by ghost** [comment]
THE CHANGED FILES
KEEP
INCREASING
AAAAAAAAAAAAAAAAAA

File: code/modules/mob/living/carbon/alien/larva/powers.dm (hunk 6 above)
Comment (ghost) [inline code/modules/mob/living/carbon/alien/larva/powers.dm]: This has been fixed.

**Comment by ghost** [comment]
Note to self, if #10234 gets merged, need to do some plane things with it.

**Comment by PJB3005** [comment]
Celt right now.

**Comment by ghost** [comment]
alright, as far as I can tell there are no obvious bugs here, but I expect there's probably a few oldcode traps that are out there somewhere that will not play kindly with it.

**Comment by ghost** [comment]
Man, don't you hate it when people use defines instead of numbers? I much prefer numbers because they're not at all maintainable and as a result cause people bringing in new features a lot of bother because the original coder couldn't be arsed searching up which define he wanted

**Comment by ComicIronic** [comment]
Sucking up to maintainers will not get this merged faster.

**Comment by ghost** [comment]
@ComicIronic I don't think you understand how much awful code I've had to trudge through to get this to work, with basically 75% of the bugs I found were due to some lazy arsehole

**Comment by ComicIronic** [comment]
> i don't you understand

L M A O

**Comment by Intigracy** [comment]
> I don't think you understand how much awful code I've had to trudge through to get this to work

But you didn't add THAT much code :^)

**Comment by ghost** [comment]
Now testing if #10234 has any adverse effects when combined with this PR.

**Comment by ghost** [comment]
It does not. The only other PR I see that would conflict is our friend #10490

**Comment by clusterfack** [comment]
747

**Comment by ghost** [comment]
thanks planes.

**Comment by ghost** [comment]
no conflicts. probably ready to merge to be honest but I'll double check

**Comment by ghost** [comment]
yeah seems so
